### PR TITLE
dcrsqlite: create DB-file parent directory when necessary

### PR DESCRIPTION
**Pull request by JFixby**
Adds test to reproduce https://github.com/decred/dcrdata/issues/515

### CODE REVIEW

**Comment by JFixby** [comment]
Fails as expected.

File: db/dcrsqlite/dcrsqlite_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		t.Fatalf("Failed to produce DB-test folder path: %v", err)
+	}
+	return testDir
+}
+
+// Ensures we run our test in a clean room. Removes all files created by any of these tests in the temp directory.
+func resetTempFolder(t *testing.T) {
+	testFolderPath := getTempFolderPath(t)
+
+	removeContents(testFolderPath)
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: unhandles error

File: db/dcrsqlite/dcrsqlite_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	err := os.RemoveAll(testFolderPath)
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: Why remove contents manually with `removeContents` when `RemoveAll` does it for you?

File: db/dcrsqlite/dcrsqlite_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	return nil
+}
+
+const (
+	defaultDataDirname = "test.data"
+	defaultDBFileName  = "test.dcrdata.sqlt.db"
+)
+
+// Returns path of a temporary directory used by these tests to store some data
+func getTempFolderPath(t *testing.T) string {
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: please remove "`get`" prefix

File: db/dcrsqlite/dcrsqlite_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	err := os.RemoveAll(testFolderPath)
+
+	//Failed to clear test-files
+	if err != nil {
+		t.Fatalf("Failed to clear temp folder %v", err)
+	}
+}
+
+// Creates a path to a file inside the temp directory
+func getFilePathInsideTempDir(t *testing.T, pathInsideTempFolder string) string {
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: Avoid starting functions with "get"

File: db/dcrsqlite/sqlite.go [hunk 5]
```diff
@@ -16,6 +16,8 @@ import (
 	apitypes "github.com/decred/dcrdata/api/types"
 	"github.com/decred/dcrdata/blockdata"
 	_ "github.com/mattn/go-sqlite3" // register sqlite driver with database/sql
+	"os"
```
Comment (chappjc) [inline db/dcrsqlite/sqlite.go]: as with other PR, segregate the std lib imports.  Put them above github stuff with a newline between the std lib and github imports

File: db/dcrsqlite/sqlite.go [hunk 6]
```diff
@@ -146,7 +148,19 @@ func NewDB(db *sql.DB) (*DB, error) {
...
 func InitDB(dbInfo *DBInfo) (*DB, error) {
-	db, err := sql.Open("sqlite3", dbInfo.FileName)
+	dbPath, err := filepath.Abs(dbInfo.FileName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensures target DB-file has a parent folder
+	parent := filepath.Dir(dbPath)
+	err = os.MkdirAll(parent, 0777)
```
Comment (chappjc) [inline db/dcrsqlite/sqlite.go]: 0755

**Comment by JFixby** [comment]
@chappjc I'm also thinking to create a package in the root of the project called smth like `testutils` and move the helpers from this test-file in there. I want to reuse it for some other tests.
Is it OK to create the new package? If not, what would be the correct way to reuse the helpers for the following tests?

**Comment by chappjc** [comment]
Yeah, a package for test helpers could be OK.

File: .gitignore [hunk 7]
```diff
@@ -30,3 +30,5 @@ MANIFEST-*
 ffldb_stake*/
 *.pprof
 testnet/
+
```
Comment (chappjc) [inline .gitignore]: remove empty line

File: db/dcrsqlite/dcrsqlite_test.go [hunk 8]
```diff
@@ -2,12 +2,13 @@ package dcrsqlite
 
 import (
 	"os"
+
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: Keep std lib imports together.

File: db/dcrsqlite/dcrsqlite_test.go [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+import (
+	"os"
+
+	"path/filepath"
+	"testing"
+)
+
+// Clears target folder content
+func RemoveContents(dir string) error {
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: now dead code

**Comment by JFixby** [comment]
Also adds some functions to read/write Block to a file.

**Comment by chappjc** [comment]
@JFixby You will have to remove that merge commit.  Use `git rebase master` when you want to get your work on top of master.  Make sure `master` refers to decred's master.

File: db/dcrsqlite/dcrsqlite_test.go (hunk 8 above)
Comment (JFixby) [reply db/dcrsqlite/dcrsqlite_test.go]: this should really be checked by CI

File: .gitignore (hunk 7 above)
Comment (JFixby) [reply .gitignore]: this should really be checked by CI

**Pull request merged**